rebasehelper/archive.py

```diff
@@ -190,42 +190,6 @@ def extract(cls, archive=None, filename=None, path=None, *args, **kwargs):
 
 
 @register_archive_type
-class SigArchiveType(ArchiveTypeBase):
-    """ .sig955015
-     files are not archives - this is a PGP file type"""
-    EXTENSION = ".sig"
-
-    @classmethod
-    def open(cls, filename=None):
-        pass
-
-    @classmethod
-    def extract(cls, archive=None, filename=None, path=None, *args, **kwargs):
-        if archive is not None:
-            raise RuntimeError("In Sig files, the archive (pos 1) argument is not used, but passed")
-        final_dir = os.path.join(path, os.path.basename(filename.rstrip(cls.EXTENSION)))
-        os.makedirs(final_dir)
-        shutil.copy(filename, final_dir)
-
-
-@register_archive_type
-class AscArchiveType(ArchiveTypeBase):
-    """ .asc files are not archives - this is a PGP file type"""
-    EXTENSION = ".asc"
-
-    @classmethod
-    def open(cls, filename=None):
-        pass
-
-    @classmethod
-    def extract(cls, archive=None, filename=None, path=None, *args, **kwargs):
-        if archive is not None:
-            raise RuntimeError("In Asc files, the archive (pos 1) argument is not used, but passed")
-        final_dir = os.path.join(path, os.path.basename(filename.rstrip(cls.EXTENSION)))
-        os.makedirs(final_dir)
-        shutil.copy(filename, final_dir)
-
-@register_archive_type
```

Comment:
Previously these were needed in order for such Source files to be downloaded from URL. The reason was bad logic in the SpecFile class method responsible for downloading Sources, because it was downloading only files with extensions of known archive types. This is not needed any more as all Sources with URL are downloaded.

.gem are different as these are listed as Source0. Arguably the handling should be improved to try to extract only files which are archives and not to add pseudo classes for such use cases.